provider/job_utils.py

```diff
@@ -51,7 +53,12 @@ def _wait_until_block_job_completed():
         if status == "pending":
             block_job_finalize(vm, job_id)
         if status == "ready":
-            block_job_complete(vm, job_id, timeout)
+            try:
+                arguments = {"id": job_id}
+                vm.monitor.cmd("job-complete", arguments)
+            except QMPCmdError as err:
+                ignore_error = err.data["desc"] == "Job not found"
+                assert ignore_error, "job-complete failed as %s" % err.data["desc"]
```

Comment:
We might don't have to care about the exact error here, in case the job status changes from READY to PENDING(if I remembered correctly, @aliang123 mentioned READY->PENDING), just log a msg here, the case will always fail due to failing to get the complete event, and then we can get the exact error msg from log.

If we want to catch the exact error msgs here, we have to consider all situations, like
1. job-complete a non-existed job
2. job-complete a pending job
Any other situations?

Reply:
Good point,  the current approach only focus on the scenario 1, complete a non-existed job.
I add this checking as i am afraid other exceptions be missed although the expected event is normal.  Before this patch, any exceptions will be raised by vm.monitor.cmd.

@aliang123 As zhenchao mentioned, is there any other situation besides the 2 above ? Thanks!

Reply:
@yanan-fu, no at present.

Reply:
Hi, @yanan-fu, have a issue here. If the "BLOCK_JOB_COMPLETED" event can't sent on time. After times of "job-complete", the "Finish" may still be "False", for" job_id not  in [data.get("id"), data.get("device")]", which will also cause case fail.
So, my suggestion is if we should add a loop check of vm.monitor.get_events for "BLOCK_JOB_COMPLETED" event ?
Example, we check if we can get the correct event in 10s. If not, we can sent "job-complete" again.

Reply:
As we discussed offline, `_wait_until_block_job_completed` response to the loop checking,  we do not need to add this step, thanks!

Reply:
@zhencliu  Discussed with @aliang123 , we all agree to remove the step of check `job-complete` exception, just send command and ignore all exceptions.   We trust the way of checking `BLOCK_JOB_COMPLETED` event, as long as no error with the event, test pass.   I willl update the code, thanks!

Reply:
hi @yanan-fu @aliang123 , I agreed with you that we don't need to check the exact error here(assertion), but I still suggest that we log a msg here, which can tell us what happens. What do you think?

Reply:
Agree, updated, and i drop `QMPCmdError` to catch all exceptions as `vm.monitor.cmd` can raise not only `QMPCmdError`. Thanks!